cmake/target.cmake

```diff
@@ -0,0 +1,21 @@
+# Copyright 2022 PingCAP, Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if (CMAKE_SYSTEM MATCHES "Linux")
+    set (ARCH_LINUX 1)
+elseif (CMAKE_SYSTEM MATCHES "FreeBSD")
+    set (ARCH_FREEBSD 1)
+elseif (CMAKE_SYSTEM_NAME MATCHES "Darwin")
+    set (ARCH_DARWIN 1)
+endif ()
```

Comment:
maybe OS_LINUX/OS_FREEBSD/OS_DARWIN is more suitable to make it distinguish from `ARCH_AMD64`/`ARCH_ARM`